instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpClientAttributesExtractor.java

```diff
@@ -141,8 +151,69 @@ private static String stripSensitiveData(@Nullable String url) {
     }
 
     if (atIndex == -1 || atIndex == len - 1) {
-      return url;
+      return redactSensitiveParameters ? redactUrlParameters(url) : url;
```

Comment:
have you considered structuring it like
```java
String stripSensitiveData(@Nullable String url) {
  if (url == null || url.isEmpty()) {
    return url;
  }
  url = redactUserInfo(url);
  url = redactQueryParameters(url);
  return url;
}
```
then you will not need to duplicate `redactSensitiveParameters ? redactUrlParameters(url) : url;`